framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,286 +85,180 @@ private static Pattern anchored(String regex) {
+    /**
+     * Converts the passed Javaparser Expression to a {@link Receiver}.
+     *
+     * @return the Receiver of the passed expression
+     */
+    private static Receiver getReceiverFromExpression(
+            Expression expr, FlowExpressionContext context, TreePath path)
             throws FlowExpressionParseException {
-        expression = expression.trim();
 
         ProcessingEnvironment env = context.checkerContext.getProcessingEnvironment();
         Types types = env.getTypeUtils();
 
-        if (isNullLiteral(expression, context)) {
-            return parseNullLiteral(types);
-        } else if (isIntLiteral(expression, context)) {
-            return parseIntLiteral(expression, types);
-        } else if (isLongLiteral(expression, context)) {
-            return parseLongLiteral(expression, types);
-        } else if (isFloatLiteral(expression, context)) {
-            throw constructParserException(
-                    expression,
-                    String.format("floating-point values '%s' cannot be parsed", expression));
-        } else if (isStringLiteral(expression, context)) {
-            return parseStringLiteral(expression, types, env.getElementUtils());
-        } else if (isThisLiteral(expression, context)) {
-            return parseThis(context);
-        } else if (isSuperLiteral(expression, context)) {
-            return parseSuper(expression, types, context);
-        } else if (isIdentifier(expression)) {
-            return parseIdentifier(expression, env, path, context);
-        } else if (isParameter(expression, context)) {
-            return parseParameter(expression, context);
-        } else if (isArray(expression)) {
-            return parseArray(expression, context, path);
-        } else if (isMethodCall(expression)) {
-            return parseMethodCall(expression, context, path, env);
-        } else if (isMemberSelect(expression)) {
-            return parseMemberSelect(expression, env, context, path);
-        } else if (isParentheses(expression, context)) {
-            return parseParentheses(expression, context, path);
-        } else {
-            String message;
-            if (expression.equals("#0")) {
-                message =
-                        "one should use \"this\" for the receiver or \"#1\" for the first formal parameter";
+        if (expr != null) {
+            if (expr.isNullLiteralExpr()) {
+                return new ValueLiteral(types.getNullType(), (Object) null);
+            } else if (expr.isIntegerLiteralExpr()) {
+                return new ValueLiteral(
+                        types.getPrimitiveType(TypeKind.INT), expr.asIntegerLiteralExpr().asInt());
+            } else if (expr.isLongLiteralExpr()) {
+                return new ValueLiteral(
+                        types.getPrimitiveType(TypeKind.LONG), expr.asLongLiteralExpr().asLong());
+            } else if (expr.isDoubleLiteralExpr()) {
+                return new ValueLiteral(
```

Comment:
Please update https://checkerframework.org/manual/#java-expressions-as-arguments so that it matches the new implementation.